[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #166007
* __->__ #165896

Audit: To prevent future issues with functools.partial or callable objects.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

anijain2305 committed:
[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build

Audit: To prevent future issues with functools.partial or callable objects.

[ghstack-poisoned]

Lucaskabela reviewed (changes requested):
Error seems legit here, lets fix then this should be good to land

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

Lucaskabela reviewed (approved):
LGTM :)

anijain2305 committed:
Update on "[dynamo][remaining] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge

anijain2305 commented:
@pytorchbot merge

anijain2305 commented:
@pytorchbot merge -f "stuck jobs"
